packages/amplify-cli/src/domain/amplify-toolkit.ts

```diff
@@ -442,4 +444,20 @@ export class AmplifyToolkit {
       require(path.join(this._amplifyHelpersDirPath, 'get-imported-auth-properties')).getImportedAuthProperties;
     return this._getImportedAuthProperties;
   }
+
+  constructor() {
+    this._cleanUpTasks = new Array();
+  }
+
+  addCleanUpTask = (task: Function) => {
```

Comment:
could be nice to define the task type as `(context: Context) => void` instead of just `Function`